src/CMakeLists.txt

```diff
@@ -350,6 +350,11 @@ else ()
         SOVERSION "${NN_ABI_VERSION}")
 endif ()
 
+# Set library outputs same as top-level project binary outputs
+set_target_properties (${PROJECT_NAME} PROPERTIES RUNTIME_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR})
+set_target_properties (${PROJECT_NAME} PROPERTIES LIBRARY_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR})
+set_target_properties (${PROJECT_NAME} PROPERTIES ARCHIVE_OUTPUT_DIRECTORY ${PROJECT_BINARY_DIR})
+
```

Comment:
@JackDunaway Aaah, I haven't thought of that solution. What I usually do for my projects is to add the runtime output directory to the `PATH` (or install the library files in a known subdirectory and add the `bin/` directory to the path, which is what I currently do to in a project where I need to build and link multiple cmake dependencies). This has the advantage of working regardless of the layout and number of targets that I create.

I would be in favor of having multiple CMakeLists. When I started using CMake, I thought that having one CMakeLists per subdirectory was dumb, but now that I manage larger projects, I kind of see the point: the main CMakeLists gets easily cluttered and harder to read the more you add to it. This is mostly a question of aethetics and readability, more than functionality.